sfge/src/main/java/com/salesforce/graph/vertex/ReferenceExpressionVertex.java

```diff
@@ -16,13 +16,19 @@ public class ReferenceExpressionVertex extends AbstractReferenceExpressionVertex
      * Presence of this vertex indicates that the reference was qualified with a 'this' reference
      */
     private final LazyOptionalVertex<ThisVariableExpressionVertex> thisVariableExpression;
+    /**
+     * Presence of this vertex indicates that the reference was qualified with a {@code super}
+     * reference.
+     */
+    private final LazyOptionalVertex<SuperVariableExpressionVertex> superVariableExpression;
 
     ReferenceExpressionVertex(Map<Object, Object> properties) {
         super(properties);
-        // TODO: Efficiency. This does 2 queries, is it safe to get the child and then look at the
+        // TODO: Efficiency. This does 3 queries, is it safe to get the child and then look at the
```

Comment:
Thought about actually trying to do the refactor for this, but decided against it. May revisit it on Tuesday, because it looks like it would probably be pretty straightforward.